src/main/java/org/apache/commons/imaging/formats/webp/chunks/WebPChunk.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.imaging.formats.webp.chunks;
+
+import org.apache.commons.imaging.ImageReadException;
+import org.apache.commons.imaging.common.BinaryFileParser;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.ByteOrder;
+import java.nio.charset.StandardCharsets;
+
+public abstract class WebPChunk extends BinaryFileParser {
+    public static final int TYPE_VP8 = 0x20385056;
+    public static final int TYPE_VP8L = 0x4C385056;
+    public static final int TYPE_VP8X = 0x58385056;
+    public static final int TYPE_ANIM = 0x4D494E41;
+    public static final int TYPE_ANMF = 0x464D4E41;
+    public static final int TYPE_ICCP = 0x50434349;
+    public static final int TYPE_EXIF = 0x46495845;
+    public static final int TYPE_XMP = 0x20504D58;
+
+    private final int type;
+    private final int size;
+    protected final byte[] bytes;
+
+    WebPChunk(int type, int size, byte[] bytes) throws ImageReadException {
+        super(ByteOrder.LITTLE_ENDIAN);
+
+        if (size != bytes.length) {
+            throw new AssertionError("Chunk size must match bytes length");
+        }
+
+        this.type = type;
+        this.size = size;
+        this.bytes = bytes;
+    }
+
+    public int getType() {
+        return type;
+    }
+
+    public String getTypeDescription() {
+        return new String(new byte[]{
+                (byte) (type & 0xff),
+                (byte) ((type >> 8) & 0xff),
+                (byte) ((type >> 16) & 0xff),
+                (byte) ((type >> 24) & 0xff)}, StandardCharsets.UTF_8);
+    }
+
+    public int getPayloadSize() {
+        return size;
+    }
+
+    public int getChunkSize() {
+        return size + 8 + ((size % 2) == 0 ? 0 : 1);
```

Comment:
This is a good example of something that's not too complicated, but would be nicer to have a short explanation why we are doing it. If it's in the spec, you can just say so :+1: